schema/workflow.json

```diff
@@ -1,33 +1,26 @@
 {
-  "$id": "https://serverlessworkflow.io/schemas/0.8/workflow.json",
+  "$id": "https://serverlessworkflow.io/schemas/0.9/workflow.json",
   "$schema": "http://json-schema.org/draft-07/schema#",
   "description": "Serverless Workflow specification - workflow schema",
   "type": "object",
   "properties": {
-    "id": {
-      "type": "string",
-      "description": "Workflow unique identifier",
-      "minLength": 1
-    },
-    "key": {
+    "name": {
       "type": "string",
-      "description": "Domain-specific workflow identifier",
+      "description": "Workflow definition unique identifier (which must be human readable)",
       "minLength": 1
     },
-    "name": {
+    "version": {
       "type": "string",
-      "description": "Workflow name",
-      "minLength": 1
+      "description": "Workflow version",
+      "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
     },
     "description": {
       "type": "string",
       "description": "Workflow description"
     },
-    "version": {
+    "key": {
       "type": "string",
-      "description": "Workflow version",
-      "minLength": 1,
-      "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
+      "description": "Expression that will be used to generate a domain-specific workflow instance identifier"
```

Comment:
I would rephrase to something like the following, to include the optional aspect of the property:

```
Optional expression used to generate a domain-specific workflow instance identifier
```

Reply:
I think the fact that key is optional is indicated in other place and feel it to be redundant while writing it, but it does not harm to emphasize it